src/pandapipes/diagnostic.py

```diff
@@ -0,0 +1,145 @@
+        logger.info(f"The pipeflow converges after {net._internal_results['iterations']:d} "
+                    f"iterations.")
+    except PipeflowNotConverged:
+        logger.info(f"After {iterations:d} iterations the pipeflow did NOT converge.")
+
+    # check with little sink and source scaling
+    logger.info("Testing with scaled-down sinks and sources.")
+    net3 = net.deepcopy()
+    if hasattr(net, "sink"):
+        net3.sink.scaling *= check_scaling_factor
+    if hasattr(net, "source"):
+        net3.source.scaling *= check_scaling_factor
+    try:
+        pp.pipeflow(net3)
+        if net3.converged:
+            logger.info(f"If sinks and sources were scaled with a factor of to "
+                        f"{check_scaling_factor}, the pipeflow would converge.")
+        else:
+            logger.warning(f"If sinks and sources were scaled with a factor of to "
+                           f"{check_scaling_factor}, the pipeflow would still NOT converge.")
+    except Exception as e:
+        logger.info(f"Pipeflow does not converge with sinks/sources scaled by"
+                    f" {check_scaling_factor}.\n"
+                    f"\t\tThe error message is: {e}")
+
+    # check k
+    if any(net.pipe.k_mm > 0.5):
+        logger.warning(f"Some pipes have a friction factor k_mm > 0.5 (extremely rough). The "
+                       f"highest value in the net is {net.pipe.k_mm.max()}. Up to "
+                       f"0.2 mm is a common value for old steel pipes."
+                       f"\nRough pipes: {net.pipe.loc[net.pipe.k_mm > 0.5]}.")
+    net4 = net.deepcopy()
+    net4.pipe.k_mm = 1e-5
+    try:
+        pp.pipeflow(net4)
+        if net4.converged:
+            logger.info(f"If the friction factor would be reduced to 1e-5 for all pipes, "
+                        f"the pipeflow would converge.")
+        else:
+            logger.warning(f"If the friction factor would be reduced to 1e-5 for all pipes, "
+                           f"the pipeflow would still NOT converge.")
+    except Exception as e:
+        logger.info(f"Pipeflow does not converge with k_mm = 1-e5 for all pipes.\n"
+                    f"\t\tThe error message is: {e}")
+
+    # check sink and source junctions:
+    node_component = ["sink", "source", "ext_grid"]
+    for nc in node_component:
+        if hasattr(net, nc):
+            missing = np.setdiff1d(net[nc].junction, net.junction.index)
+            if len(missing):
+                logger.warning(f"Some {nc}s are connected to non-existing junctions!"
+                               f"\n{nc}s:{net[nc].loc[net[nc].junction.isin(missing)]}"
+                               f"\nmissing junctions:{missing}")
+
+    # check from and to junctions
+    branch_component = ["pipe", "valve", "compressor", "pump", "heat_exchanger", "circulation_pump"]
+    for bc in branch_component:
+        if hasattr(net, bc):
+            missing_f = np.setdiff1d(net[bc].from_junction, net.junction.index)
```

Comment:
I would prefer the following:
1. Add some code above that checks all tables in net and makes sure that the respective model is part of the component list (an important source of errors).
2. Iterate over all branch components and use the "from_to_junction" identifiers to check the columns